lib/ultrajsonenc.c

```diff
@@ -941,12 +941,11 @@ char *JSON_EncodeObject(JSOBJ obj, JSONObjectEncoder *enc, char *_buffer, size_t
 
   encode (obj, enc, NULL, 0);
 
-  Buffer_Reserve(enc, 1);
   if (enc->errorMsg)
   {
     return NULL;
   }
-  Buffer_AppendCharUnchecked(enc, '\0');
```

Comment:
I'd say that they got what was coming to them if something did break because of it.

Reply:
This might very well be wrong, but my understanding is that this is an internal interface, and only the Python methods are considered public API.

Assuming that's correct... Anyone directly (ab)using internal interfaces should expect them to be changed at any time without warning and things to blow up. And I'd rather not do useless work for a weird edge case that itself causes some performance edge cases (namely, if the NUL byte causes a buffer resizing).